apps/server/src/modules/user-login-migration/uc/user-login-migration.uc.ts

```diff
@@ -36,4 +49,86 @@ export class UserLoginMigrationUc {
 			await this.userLoginMigrationService.findUserLoginMigrations(query, options);
 		return userLoginMigrations;
 	}
+
+	async migrate(
+		userJwt: string,
+		currentUserId: string,
+		targetSystemId: EntityId,
+		code: string,
+		redirectUri: string
+	): Promise<void> {
+		const tokenDto: OAuthTokenDto = await this.oauthService.authenticateUser(targetSystemId, redirectUri, code);
+
+		this.logMigrationInformation(currentUserId, `Migrates to targetSystem with id ${targetSystemId}`);
+
+		const data: OauthDataDto = await this.provisioningService.getData(
+			targetSystemId,
+			tokenDto.idToken,
+			tokenDto.accessToken
+		);
+
+		this.logMigrationInformation(currentUserId, undefined, data, targetSystemId);
+
+		if (data.externalSchool) {
+			const schoolToMigrate: SchoolDO | null = await this.schoolMigrationService.schoolToMigrate(
+				currentUserId,
+				data.externalSchool.externalId,
+				data.externalSchool.officialSchoolNumber
+			);
+
+			this.logMigrationInformation(
+				currentUserId,
+				`Found school with officialSchoolNumber (${data.externalSchool.officialSchoolNumber ?? ''})`
+			);
+
+			if (schoolToMigrate) {
+				await this.schoolMigrationService.migrateSchool(
+					data.externalSchool.externalId,
+					schoolToMigrate,
+					targetSystemId
+				);
+
+				this.logMigrationInformation(currentUserId, undefined, data, data.system.systemId, schoolToMigrate);
+			}
+		}
+
+		const migrationDto: MigrationDto = await this.userMigrationService.migrateUser(
+			currentUserId,
+			data.externalUser.externalId,
+			targetSystemId
+		);
+
+		// TODO: N21-820 when the userMigrationService.migrateUser will not longer be used throw errors directly
+		if (migrationDto.redirect.includes('migration/error')) {
+			throw new InternalServerErrorException();
```

Comment:
can you please return an error that makes sense here. we also need to now, if this is the school number mismatch.

Reply:
done